bin/crew

```diff
@@ -844,10 +845,12 @@ def pre_install(dest_dir)
   end
 end
 
-def post_install(dest_dir)
-  Dir.chdir dest_dir do
-    puts 'Performing post-install...'
-    @pkg.postinstall
+def post_install
+  Dir.mktmpdir do |post_install_tempdir|
+    Dir.chdir post_install_tempdir do
+      puts "Performing post-install for #{@pkg.name}...".lightblue
+      @pkg.postinstall
+    end
   end
```

Comment:
What is the justification for running postinstall in a temporary directory?

Reply:
post_install was previously being passed CREW_DEST_DIR, and the contents of that could change between package installs. Since postinstall is now serialized, one can not assume that CREW_DEST_DIR will stay the same between running the package extract and the postinstall, so this makes it explicit that post_install does not have access to the package extraction CREW_DEST_DIR.

Reply:
This has been an issue for a long time but wouldn't it make sense to clear out `CREW_DEST_DIR` after each package is processed?  I think you need to remove the temp dir after processing also, correct?  When installing packages that have a lot of dependencies, it can potentially require a lot of disk space if it isn't cleared out each time, right?

Reply:
I can see a potential problem with delaying postinstalls until the end.  Some packages may "depend" on a postinstall process as a prerequisite.

Reply:
The temp directory is automatically removed at the end of the function block.

CREW_DEST_DIR is only used for builds, yes?

That CREW_DEST_DIR is what post_install was being pointed to is its own issue, which is now addressed.

I think we are talking about `CREW_BREW_DIR` though, and that is handled already by 
```
    # cleanup
    unless @opt_keep
      FileUtils.rm_rf Dir.glob("#{CREW_BREW_DIR}/*")
      FileUtils.mkdir_p "#{CREW_BREW_DIR}/dest" # this is a little ugly, feel free to find a better way
    end
```

`@extract_dir` appears to be the location for packages being extracted, 

(One could also do the same for CREW_BREW_DIR, by using the tempdir function for that, but that might get hairy since we need to make sure that the mktempdir function block encompasses all use of that folder.)

Maybe that's a separate PR?

Reply:
Is the idea behind delaying the postinstalls is so the messages don't scroll by as other dependencies are installed?  If that is the only reason, we could create a new section just for messages.  Would something like `def self.messages` work?

Reply:
The primary reason behind delaying postinstalls is to make sure all binaries from dependencies are installed before they can be invoked from postinstall sections of packages. This way we avoid issues with worrying about whether for instance an icon cache for `geany` can be updated by `gtk-update-icon-cache` before `gtk-update-icon-cache` is available from having installed the `gtk3` package.

Having postinstall messages all show up at the end is just a side benefit.